drivers/misc/nxp_lp_flexcomm/nxp_lp_flexcomm.c

```diff
@@ -0,0 +1,169 @@
+ */
+
+#define DT_DRV_COMPAT nxp_lp_flexcomm
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include "nxp_lp_flexcomm.h"
+
+LOG_MODULE_REGISTER(nxp_lp_flexcomm, LOG_LEVEL_ERR);
+
+struct nxp_lp_flexcomm_child {
+	const struct device *dev;
+	uint8_t periph;
+	child_isr_t lp_flexcomm_child_isr;
+};
+
+struct nxp_lp_flexcomm_data {
+	struct nxp_lp_flexcomm_child *children;
+	size_t num_children;
+};
+
+struct nxp_lp_flexcomm_config {
+	LP_FLEXCOMM_Type *base;
+	void (*irq_config_func)(const struct device *dev);
+};
+
+void nxp_lp_flexcomm_isr(const struct device *dev)
+{
+	uint32_t interrupt_status;
+	const struct nxp_lp_flexcomm_config *config = dev->config;
+	struct nxp_lp_flexcomm_data *data = dev->data;
+	uint32_t instance = LP_FLEXCOMM_GetInstance(config->base);
+	struct nxp_lp_flexcomm_child *child;
+
+	interrupt_status = LP_FLEXCOMM_GetInterruptStatus(instance);
+	if ((interrupt_status &
+	    ((uint32_t)kLPFLEXCOMM_I2cSlaveInterruptFlag |
+	     (uint32_t)kLPFLEXCOMM_I2cMasterInterruptFlag)) != 0U) {
+		child = &data->children[LP_FLEXCOMM_PERIPH_LPI2C];
+
+		if (child->lp_flexcomm_child_isr != NULL) {
+			child->lp_flexcomm_child_isr(child->dev);
+		}
+	}
+	if ((interrupt_status &
+	     ((uint32_t)kLPFLEXCOMM_UartRxInterruptFlag |
+	      (uint32_t)kLPFLEXCOMM_UartTxInterruptFlag)) != 0U) {
+		child = &data->children[LP_FLEXCOMM_PERIPH_LPUART];
+
+		if (child->lp_flexcomm_child_isr != NULL) {
+			child->lp_flexcomm_child_isr(child->dev);
+		}
+	}
+	if (((interrupt_status &
+	     (uint32_t)kLPFLEXCOMM_SpiInterruptFlag)) != 0U) {
+		child = &data->children[LP_FLEXCOMM_PERIPH_LPSPI];
+
+		if (child->lp_flexcomm_child_isr != NULL) {
+			child->lp_flexcomm_child_isr(child->dev);
+		}
+	}
+}
+
+void nxp_lp_flexcomm_setirqhandler(const struct device *dev, const struct device *child_dev,
+				   LP_FLEXCOMM_PERIPH_T periph, child_isr_t handler)
+{
+	struct nxp_lp_flexcomm_data *data = dev->data;
+	struct nxp_lp_flexcomm_child *child;
+
+	child = &data->children[periph];
+
+	/* Store the interrupt handler and the child device node */
+	child->lp_flexcomm_child_isr = handler;
+	child->dev = child_dev;
+}
+
+static int nxp_lp_flexcomm_init(const struct device *dev)
+{
+	const struct nxp_lp_flexcomm_config *config = dev->config;
+	struct nxp_lp_flexcomm_data *data = dev->data;
+	uint32_t instance;
+	struct nxp_lp_flexcomm_child *child = NULL;
+	bool spi_found = false;
+	bool uart_found = false;
+	bool i2c_found = false;
+
+	for (int i = 1; i < data->num_children; i++) {
+		child = &data->children[i];
+		if (child->periph == LP_FLEXCOMM_PERIPH_LPSPI) {
+			spi_found = true;
+		}
+		if (child->periph == LP_FLEXCOMM_PERIPH_LPI2C) {
+			i2c_found = true;
+		}
+		if (child->periph == LP_FLEXCOMM_PERIPH_LPUART) {
+			uart_found = true;
+		}
+	}
+
+	/* If SPI is enabled with another interface type return an error */
+	if (spi_found && (i2c_found || uart_found)) {
```

Comment:
a lot of this init function should be able to be determined at build time as well I think

Reply:
I cannot think of a way to cover this logic at build time.